modules/core/src/lib/timeline.js

```diff
@@ -0,0 +1,94 @@
+const WRAP_LOOP = 0;
+const WRAP_CLAMP = 1;
+const WRAP_MAP = {
+  loop: WRAP_LOOP,
+  clamp: WRAP_CLAMP
+};
+
+export class Timeline {
+  constructor() {
+    this.time = 0;
+    this.duration = Number.POSITIVE_INFINITY;
+    this.wrapMode = WRAP_LOOP;
+    this.channels = [];
+    this.rate = 1;
+    this.playing = false;
+    this.lastEngineTime = -1;
+  }
+
+  addChannel(props) {
```

Comment:
Thinking about using this for deck.gl's animation features -- we'll also need the ability to remove a channel when an animation is discarded. Which means using the array index as `handle` will be problematic.

Reply:
No problem. I'll switch to a Map and add a `removeChannel` method.